mesonbuild/compilers/compilers.py

```diff
@@ -445,6 +446,9 @@ def __init__(self, compiler: T.Union['Compiler', StaticLinker],
                  iterable: T.Optional[T.Iterable[str]] = None):
         self.compiler = compiler
         self.__container = list(iterable) if iterable is not None else []  # type: T.List[str]
+        self.__fast_loopup = {}
+        for arg in self.__container:
+            self.__fast_loopup[arg] = True
```

Comment:
why don't we just replace `self.__container` with a `mesonlib.OrderedDict`, so we can get the log(n) lookup, but not have to keep two containers in sync?

Reply:
Mhm, i tried to give this a try. However, i cannot implement then `__getitem__` `__setitem__` etc.
But as i said in my commit for this, this is not the end, there can be way more to improve this, maybe without two containers, but a other data struct.

Reply:
Okay. I'm still not crazy about having two data structures, but we probably need a different datastructure, maybe something like a trie? But that's for another time I guess.

Reply:
Yeah - i totally agree on that, I will not have much time in the next few months, i hope i can revisit this, and build the structure i was thinking about.